tests/test_geometry_3d.h

```diff
@@ -0,0 +1,417 @@
+	tt.push_back(Case(3.0f, 10.0f, 10, Vector3::Axis(), 12));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case currentCase = tt[i];
+		Vector<Plane> planes = Geometry3D::build_cylinder_planes(currentCase.Radius, currentCase.Height, currentCase.Sides, currentCase.Axis);
+		CHECK(planes.size() == currentCase.WantSize);
+	}
+}
+TEST_CASE("[Geometry3D] Build Sphere Planes") {
+	struct Case {
+		real_t Radius;
+		int Lats, Lons;
+		Vector3::Axis Axis;
+		int WantSize;
+		Case(){};
+		Case(real_t radius, int lat, int lons, Vector3::Axis axis, int wantSize) :
+				Radius(radius), Lats(lat), Lons(lons), Axis(axis), WantSize(wantSize){};
+	};
+	Vector<Case> tt;
+	tt.push_back(Case(10.0f, 10, 3, Vector3::Axis(), 63));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case currentCase = tt[i];
+		Vector<Plane> planes = Geometry3D::build_sphere_planes(currentCase.Radius, currentCase.Lats, currentCase.Lons, currentCase.Axis);
+		CHECK(planes.size() == 63);
+	}
+}
+TEST_CASE("[Geometry3D] Build Convex Mesh") {
+	struct Case {
+		Vector<Plane> Object;
+		int WantFaces, WantEdges, WantVertices;
+		Case(){};
+		Case(Vector<Plane> object, int wantFaces, int wantEdges, int wantVertices) :
+				Object(object), WantFaces(wantFaces), WantEdges(wantEdges), WantVertices(wantVertices){};
+	};
+	Vector<Case> tt;
+	tt.push_back(Case(Geometry3D::build_box_planes(Vector3(5, 10, 5)), 6, 12, 8));
+	tt.push_back(Case(Geometry3D::build_capsule_planes(5, 5, 20, 20, Vector3::Axis()), 820, 7603, 6243));
+	tt.push_back(Case(Geometry3D::build_cylinder_planes(5, 5, 20, Vector3::Axis()), 22, 100, 80));
+	tt.push_back(Case(Geometry3D::build_sphere_planes(5, 5, 20), 220, 1011, 522));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case currentCase = tt[i];
+		Geometry3D::MeshData mesh = Geometry3D::build_convex_mesh(currentCase.Object);
+		CHECK(mesh.faces.size() == currentCase.WantFaces);
+		CHECK(mesh.edges.size() == currentCase.WantEdges);
+		CHECK(mesh.vertices.size() == currentCase.WantVertices);
+	}
+}
+TEST_CASE("[Geometry3D] Clip Polygon") {
+	struct Case {
+		Plane ClippingPlane;
+		Vector<Vector3> Polygon;
+		bool WantEqual;
+		Case(){};
+		Case(Plane clippingPlane, Vector<Vector3> polygon, bool wantEqual) :
+				ClippingPlane(clippingPlane), Polygon(polygon), WantEqual(wantEqual){};
+	};
+	Vector<Case> tt;
+	Vector<Plane> boxPlanes = Geometry3D::build_box_planes(Vector3(5, 10, 5));
+	Vector<Vector3> box = Geometry3D::compute_convex_mesh_points(&boxPlanes[0], boxPlanes.size());
+	tt.push_back(Case(Plane(), box, true));
+	tt.push_back(Case(Plane(Vector3(0, 3, 0), Vector3(0, 1, 0)), box, false));
+	for (int i = 0; i < tt.size(); ++i) {
+		Case currentCase = tt[i];
+		Vector<Vector3> output = Geometry3D::clip_polygon(currentCase.Polygon, currentCase.ClippingPlane);
+		if (currentCase.WantEqual) {
+			CHECK(output == currentCase.Polygon);
+		} else {
+			CHECK(output != currentCase.Polygon);
+		}
+	}
+}
+TEST_CASE("[Geometry3D] Compute Convex Mesh Points") {
+	struct Case {
+		Vector<Plane> Mesh;
```

Comment:
```suggestion
		Vector<Plane> mesh;
```
For parameters, `p_mesh` can be used, which is again the naming convention Godot currently follows.